k-frontend/src/main/java/org/kframework/parser/outer/ParseRegex.java

```diff
@@ -124,6 +124,26 @@ public static Regex parse(String regex) {
     return parse(new Scanner(regex));
   }
 
+  public static RegexBody parseNamed(String regex, HasLocation loc) {
+    return parseNamed(new Scanner(regex, loc));
+  }
+
+  public static RegexBody parseNamed(String regex) {
+    return parseNamed(new Scanner(regex));
+  }
+
+  private static RegexBody parseNamed(Scanner input) {
+    Regex reg = parse(input);
+    if (reg.startLine() || reg.endLine()) {
+      String msg = "Named lexical syntax cannot contain line anchors.";
```

Comment:
Can you add this to the test? It looks like the location info is generally lost here as well, any idea why?

Reply:
Do you have an example where the location info gets lost? It seems to work correctly on a simple test `syntax lexical StartAnchor = r"^foo"`.

Reply:
My bad. I added the example to `checkLexicalIdentifiers.k` and ran the target, and misread the diff. The location info was there.